adds Tank Autocannon

Pariah919 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Adds the Bushwacker Atuocannon to the Tank. You can't actually get it since I don't know where tivi put all the spawn stuff and I haven't reinstalled SDMM.

Works like how you'd expect a autocannon to function. Has AP and HE. Fires slower than LTAAP and doesn't have IFF.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
More tank gun variety.
<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to manipulate the system all you want, but it's best to shoot for clear communication right off the bat. -->

<!--Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->

<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents, don't be too verbose. -->

:cl:
add: Added new tank Autocannon. Has two ammo types- HE and AP. It fires faster than LTAAP and does not have IFF.
/:cl:

Lumipharon commented on code/modules/projectiles/ammo_types/heavy_ammo.dm:
```diff
@@ -183,6 +183,40 @@
 
+// Tank Autocannon
+
+/datum/ammo/bullet/tank_autocannon
+	name = "autocannon armor piercing"
+	hud_state = "alloy_spike"
+	hud_state_empty = "smartgun_empty"
+	ammo_behavior_flags = AMMO_BALLISTIC
+	damage = 30
+	penetration = 40
+	sundering = 2.5
+
+/datum/ammo/bullet/tank_autocannon
+	name = "autocannon high explosive"
+	damage = 15
+	penetration = 20
+	sundering = 1.5
+
+/datum/ammo/bullet/tank_autocannon/high_explosive/drop_nade(turf/T)
+	explosion(T, light_impact_range = 1, weak_impact_range = 3)
+
+/datum/ammo/bullet/tank_autocannon/high_explosive/on_hit_mob(mob/target_mob, obj/projectile/proj)
+	var/target_turf = get_turf(target_mob)
+	staggerstun(target_mob, proj, max_range, knockback = 1, hard_size_threshold = 3)
+	drop_nade(target_turf)
+
+/datum/ammo/bullet/tank_autocannon/high_explosive/on_hit_obj(obj/target_obj, obj/projectile/proj)
+	drop_nade(target_obj.density ? get_step_towards(target_obj, proj) : target_obj.loc)
+
+/datum/ammo/bullet/tank_autocannon/high_explosive/on_hit_turf(turf/target_turf, obj/projectile/proj)
+	drop_nade(target_turf.density ? get_step_towards(target_turf, proj) : target_turf)
+
+/datum/ammo/bullet/tank_autocannon/high_explosive/do_at_max_range(turf/target_turf, obj/projectile/proj)
+	drop_nade(target_turf.density ? get_step_towards(target_turf, proj) : target_turf)
```
probably easier to make this a subtype of rocket and just change the flags tbh, as this is a lot of copy paste

Lumipharon commented on code/modules/vehicles/armored/ammo_magazine.dm:
```diff
@@ -52,6 +52,20 @@
+	desc = "A 200 round box for an autocannon. Loaded with Armor Piercing rounds."
+	caliber = CALIBER_30X17MM
+	icon_state = "tank_autocannon_ap"
+	max_rounds = 250
+	default_ammo = /datum/ammo/bullet/tank_autocannon
```
gonna leave this to kuro but uh, xenos are sure gonna have an interesting time against 250 grenade magazines that can be rapid fired

Pariah919 replied:
Gonna rework it into just firing fragmentation 360. I put down high explosive as explosions since that was the first thing that came to mind

TiviPlus commented:
no damaged sprite btw

Kuro020 reviewed (changes requested):
HE staggers,slows and knockbacks, thats 2 much i think

Also the ammo economy is whack, 250 per mag and you get 20 free mags at the start, pls nerf

Pariah919 commented:
>HE staggers,slows and knockbacks, thats 2 much i think

I removed the inherited knockback. Slow is from weak explosions. The stagger is people coping though so I can't remove what's already not there.

>Also the ammo economy is whack, 250 per mag and you get 20 free mags at the start, pls nerf

Dropped to 100.
